prov/efa/src/efa_hmem.c

```diff
@@ -142,6 +134,23 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 	else
 		info->p2p_required_by_impl = true;
 
+	if (ofi_hmem_p2p_disabled()) {
+		info->p2p_disabled_by_user = true;
+		info->p2p_supported_by_device = false;
+		efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_CUDA);
```

Comment:
Doesn't really make sense to initialize the protocol thresholds based on the condition of disabled P2P - just move the call from further down the function above the `ofi_hmem_p2p_disabled()` and make the init unconditional.

Reply:
Without initializing `p2p_supported_by_device`, we will always fall back to FI_HMEM_SYSTEM initialization logic if we move `efa_domain_hmem_info_init_protocol_thresholds` above. That is not what we expect.